src/test/java/assertions/PersonTest.java

```diff
@@ -10,30 +12,39 @@
 
     @Test
     void getFullNameReturnsFirstnameSpaceLastname(){
-        // TODO implement
-        throw new IllegalArgumentException("you should implement code here");
+        Person p = new Person("Urs", "Heusser", LocalDate.of(2000,1,15));
+        String s = "Urs Heusser";
+        Assertions.assertEquals(s, p.getFullName());
     }
 
-    // TODO some more useful tests
+    @Test
+    void getFullNameHasASpace() {
+        Person p = new Person("Urs", "Heusser", LocalDate.of(2000,1,15));
+        String fn = p.getFullName();
+        Assertions.assertTrue(fn.contains(" "));
+    }
 
 
     // --- getAge
 
     @Test
-    void getAgeReturns10YearsIfBornIn2009() throws Exception {
-        // TODO verbessern. Hinweis: Repeatable (wiederholbar) zu jeder Zeit.
-        Person p = new Person("", "", LocalDate.of(2009, 1, 1));
-
-        throw new IllegalArgumentException("you should implement code here");
+    void getAgeReturns10YearsIfBornIn2009()  {
+        Person p = new Person("", "", LocalDate.now().minusYears(10));
+        Period age = p.getAge();
+        Assertions.assertEquals(10, age.getYears());
     }
 
     @Test
-    void getAgeReturns1DayIfYesterday() throws Exception {
+    void getAgeReturns1DayIfYesterday()  {
         Person p = new Person("", "", LocalDate.now().minusDays(1));
-
-        // TODO implement
-        throw new IllegalArgumentException("you should implement code here");
+        Period age = p.getAge();
+        Assertions.assertEquals(1, age.getDays());
```

Comment:
Was wäre wenn `getMonths() == 1` ..., würdest du das merken? Bzw. nur weil getDays() == 1, heisst ja nicht, dass `age = 1 Tag`

Reply:
Wahrscheinlich verstehe ich hier etwas nicht ganz. Würde in der Zeile 41 statt age.getDays() age.getMonth() eingesetzt, wäre dieser Test falsch, da ja das Geburtsdatum der Person vor einem Tag und nicht vor einem Monat war - ergo wäre der Wert age.getMonths()  in meinem Testszenario 0 und nicht 1.
<img width="722" alt="getMonths" src="https://user-images.githubusercontent.com/89865261/140368595-a71cad8a-0a1e-4ecd-87fd-5f15c75c3bde.PNG">

Reply:
Du verstehst das schon richtig. getMonths() sollte 0 sein. Du könntest das einfach noch in einer weitere Assertion sicherstellen :)